File: scope106/src/main/java/com/example/scope106/task1/ExampleLooperThread.java
package com.example.scope106.task1;

import android.content.Context;
import android.os.Handler;
import android.os.Looper;
import android.util.Log;

import com.example.scope106.task1.ExampleHandler;

public class ExampleLooperThread extends Thread {
    private static final String TAG = "ExampleLooperThread";

    public Looper looper;
    public Handler handler;
    private final Context context;

    public ExampleLooperThread(Context context) {
        this.context = context;
    }

    @Override
    public void run() {
        Looper.prepare();

        looper = Looper.myLooper();

        handler = new ExampleHandler(context);

        Looper.loop();

        Log.d(TAG, "End of run()");
    }
}

File: scope106/src/main/java/com/example/scope106/task1/ExampleLooperThread.kt
package com.example.scope106.task1

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.util.Log

class ExampleLooperThread(private val context: Context) : Thread() {
    var looper: Looper? = null
    var handler: Handler? = null
    override fun run() {
        Looper.prepare()
        looper = Looper.myLooper()
        handler = ExampleHandler(context)
        Looper.loop()
        Log.d(TAG, "End of run()")
    }

    companion object {
        private const val TAG = "ExampleLooperThread"
    }
}